### ISSUE TITLE
Switching between multiple different printers causes slicing errors

### ISSUE THREAD

**Issue by N720** [daid/LegacyCura#1453]
Switching between printers in the Machine menu in Cura 15.04.4 Windows 7 appears to cause problems with slicing.  The attached image is of a gcode file re-imported into Cura after Cura sliced it.  Printer profiles I have are Printrbot Simple Metal, Printrbot Metal Plus, and Witbox.  It has happened on gcode exports for both the Witbox and the Printrbot Simple Metal. 

Also attached is the gcode file as txt

![capture](https://cloud.githubusercontent.com/assets/18563264/14659480/421cec7a-0651-11e6-8ae7-3ead2300afe7.JPG)
[NeckV1_gcode.txt](https://github.com/daid/Cura/files/227039/NeckV1_gcode.txt)

**Comment by daid** [comment]
This is a different bug, the gcode is fine (load it in a different viewer). The bug is that the loader and the viewer are not properly synced causing problems that not everything is rendered.

**Comment by N720** [comment]
Thank you for checking out the gcode.  Here's an example of how the prints turn out.  My process is to export gcode from Cura, import to Octoprint, and print.  Must be a problem with Octoprint.  The prints work fine if I upload the stl to Octoprint and have it slice with the Cura engine.  I prefer using the Cura software however.

I've made hundreds of prints with my regular Cura -> Octoprint -> Print before; it's only when I added additional printers that the problem showed up.

![2016-04-20 18 59 40](https://cloud.githubusercontent.com/assets/18563264/14698917/05306754-0748-11e6-8b69-4b92199fb789.jpg)
